src/player.cpp

```diff
@@ -7737,35 +7737,39 @@ bool player::can_wear( const item& it, bool alert ) const
     return true;
 }
 
-bool player::can_wield( const item &it, bool alert ) const
+bool player::can_wield( const item &it, std::string *err ) const
```

Comment:
Again the "smelly" argument pass. Extra risky, because it replaces a `bool` with a pointer, meaning old invocations can be implicit-cast into new ones.

Could you tidy it up a bit before this PR lands? You said something about making a structure that would contain both the result and error reason. That would sure help to avoid both the risky signature change and C-ish optional-argument-by-pointer.

Reply:
See #21774 (the latter PR takes precedence).